### PR TITLE
Add tests configuration for StripeConnect.

### PR DESCRIPTION
**Pull request by nschris-stripe**
## Summary
Follow up to #3868. This PR adds StripeCore as a dependency to StripeConnect and configures tests for StripeConnect.

https://jira.corp.stripe.com/browse/MXMOBILE-2439

### CODE REVIEW

File: StripeConnect/StripeConnect.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8,6 +8,9 @@
 
 /* Begin PBXBuildFile section */
 		41A2A5622C5A972A0077FC74 /* Placeholder.swift in Sources */ = {isa = PBXBuildFile; fileRef = 41A2A5612C5A972A0077FC74 /* Placeholder.swift */; };
+		41A2A5642C5ABD6C0077FC74 /* StripeConnectTests.xctestplan in Resources */ = {isa = PBXBuildFile; fileRef = 41A2A5632C5ABD6C0077FC74 /* StripeConnectTests.xctestplan */; };
+		41A2A5682C5AC5120077FC74 /* StripeCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 41A2A5672C5AC5120077FC74 /* StripeCore.framework */; };
+		41D17A452C5A73A6007C6EE6 /* StripeConnect.docc in Sources */ = {isa = PBXBuildFile; fileRef = 41D17A442C5A73A6007C6EE6 /* StripeConnect.docc */; };
```
Comment (mludowise-stripe) [inline StripeConnect/StripeConnect.xcodeproj/project.pbxproj:13]: Looks like the .docc got added back to the project file

File: fastlane/Fastfile [hunk 2]
```diff
@@ -74,7 +80,11 @@ platform :ios do
 
   private_lane :legacy_tests do |options|
     # TODO(bmelts): would be nice to read these from modules.yaml
-    schemes = %w[StripeiOS StripePayments StripePaymentsUI StripePaymentSheet StripeCameraCore StripeCore StripeUICore StripeApplePay StripeCardScan]
+    schemes = %w[StripeiOS StripePayments StripePaymentsUI StripePaymentSheet StripeCameraCore StripeCore StripeUICore
+                 StripeApplePay StripeCardScan]
+
+    schemes << 'StripeConnect' if Gem::Version.new(options[:version]) >= Gem::Version.new('15.0')
```
Comment (mludowise-stripe) [inline fastlane/Fastfile:86]: ❤️

**Review by mludowise-stripe — APPROVED** [review]
1 minor fix, otherwise lgtm

**Review by porter-stripe — APPROVED** [review]

File: StripeConnect/StripeConnect.xcodeproj/project.pbxproj (hunk 1 above)
Comment (nschris-stripe) [reply StripeConnect/StripeConnect.xcodeproj/project.pbxproj:13]: Good catch, weirdly it doesn't actually show in the file navigator. I've got a follow up PR, coming and I'll delete it there.

**Pull request merged**